rfc/2025-07-22-conditional-enable.md

```diff
@@ -0,0 +1,195 @@
+2. Support for conditional enabling on:
+  1. Resources - Add a new field on the `lifecycle` block called `enabled`;
+  2. Modules - Add a new `lifecycle` block for modules but only supporting the new `enabled` field.
+
+
+## Technical Approach
+
+### `lifecycle` support on Modules
+
+The lifecycle block is currently not supported on `modules`, but in our docs we mention to users that it's reserved for future usage:
+
+> OpenTofu does not use the lifecycle argument. However, the lifecycle block is reserved for future versions.
+
+https://opentofu.org/docs/language/modules/syntax/#meta-arguments
+
+It would be the right timing to start supporting it, but only for the `enabled` field.
+
+## Migration from existing resources
+
+### Migration from `count` to `enabled`
+
+As mentioned before, users were adding support for enabling/disabling resources by using `count = var.enabled ? 1 : 0` and this is the most common case where they want to use an `enabled` field.
+Luckily, this support is [in-place already](https://github.com/opentofu/opentofu/pull/3066#discussion_r2231518392). When using a `count` for this conditional dependency, if
+the resource is changed to use `enabled`, an implicit `move` will be done on the apply phase to remove the index and
+turn it into a single instance.
+
+```
+OpenTofu will perform the following actions:
+
+  # null_resource.example[0] has moved to null_resource.example
+    resource "null_resource" "example" {
+        id = "8975736315378968412"
+    }
+
+Plan: 0 to add, 0 to change, 0 to destroy.
+```
+
+### Migration from `for_each` to `enabled`
+
+Unlike `count`, implicit moves are not handled by OpenTofu. This means that the user should explicitly
+tell OpenTofu what needs to be moved:
+
+```
+moved {
+  from = aws_s3_bucket.example["for_each_key"]  # name of the key being used by for_each
+  to = aws_s3_bucket.example
+}
+```
+
+So a complete example of that support would be:
+
+```
+# For_each before:
+
+locals {
+  buckets = ["bucket-1", "bucket-2"]
+}
+
+moved {
+  from = aws_s3_bucket.example["bucket-1"]  # name of the key being used by for_each
+  to = aws_s3_bucket.example
+}
+
+resource "aws_s3_bucket" "example" {
+  for_each = var.enable ? toset(local.buckets) : []
+  bucket = each.key
+}
+
+# To:
+resource "aws_s3_bucket" "example" {
+  lifecycle {
+    enabled = var.enable
+  }
+}
+```
+
+It doesn't make a lot of sense in the example above to move from `for_each` to `enabled` if more than one resource is being used. `bucket-2` in the example above would be removed. This path of migration
+should be used *only* if a single instance is being used and for_each is supporting that conditional.
+
+### Usage of `enabled` together with `for_each` and `count`.
+
+These three different arguments would behave similarly, but with different semantics.
+It could be argued that the user wants to enable a resource with 4 instances, but in our current code,
+that conditional can be changed to `count=0`.
```

Comment:
This makes no sense to me, can you provide some more explanation here or even some examples?